spring-pulsar-spring-boot-autoconfigure/src/test/java/org/springframework/pulsar/autoconfigure/AuthParameterUtilsTests.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.autoconfigure;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Tests for {@link AuthParameterUtils}.
+ *
+ * @author Alexander Preuß
+ */
+public class AuthParameterUtilsTests {
+
+	@ParameterizedTest(name = "{0}")
+	@MethodSource("encodedParamStringConversionProvider")
+	void encodedParamStringConversion(String testName, Map<String, String> authParamsMap) {
+		String encodedAuthParamString = AuthParameterUtils.maybeConvertToEncodedParamString(authParamsMap);
+		if (authParamsMap == null || authParamsMap.isEmpty()) {
+			assertThat(encodedAuthParamString).isNull();
+		}
+		else {
+			assertThat(encodedAuthParamString).isEqualTo("{\"audience\":\"urn:sn:pulsar:abc:xyz\","
+					+ "\"issuerUrl\":\"https://auth.server.cloud\",\"privateKey\":\"file://Users/xyz/key.json\"}");
+		}
+	}
+
+	private static Stream<Arguments> encodedParamStringConversionProvider() {
+		return Stream.of(arguments("null", null), arguments("empty", Collections.emptyMap()),
+				arguments("camelCase",
+						Map.of("issuerUrl", "https://auth.server.cloud", "privateKey", "file://Users/xyz/key.json",
+								"audience", "urn:sn:pulsar:abc:xyz")),
+				arguments("kebabCase",
+						Map.of("issuer-url", "https://auth.server.cloud", "private-key", "file://Users/xyz/key.json",
+								"audience", "urn:sn:pulsar:abc:xyz")),
+				arguments("lowerCase", Map.of("issuerurl", "https://auth.server.cloud", "privatekey",
+						"file://Users/xyz/key.json", "audience", "urn:sn:pulsar:abc:xyz")));
+	}
```

Comment:
Wdy about adding a test that uses the mechanism I described [here](https://github.com/spring-projects-experimental/spring-pulsar/pull/97#issuecomment-1242223676) which would test the env var case end-end.  I think this would require SpringBootTest in order to do the relaxed env var bindings and that does not exist in this module. Would need to do this test in the autoconfigure module. This is fine for now - I may explore doing that later.

Reply:
Sure, let's add it once your PR is merged so we can reuse the test class :)

Reply:
I created https://github.com/spring-projects-experimental/spring-pulsar/issues/112 as a follow on